modules/authentication/src/handlers/magic.link.ts

```diff
@@ -0,0 +1,151 @@
+import { isNil } from 'lodash';
+import { TokenType } from '../constants/TokenType';
+import { v4 as uuid } from 'uuid';
+import ConduitGrpcSdk, {
+  ConduitRouteActions,
+  ConduitRouteReturnDefinition,
+  ConduitString,
+  ConfigController,
+  Email,
+  GrpcError,
+  ParsedRouterRequest,
+  RoutingManager,
+  UnparsedRouterResponse,
+} from '@conduitplatform/grpc-sdk';
+import { Token, User } from '../models';
+import { status } from '@grpc/grpc-js';
+import { IAuthenticationStrategy } from '../interfaces/AuthenticationStrategy';
+import { TokenProvider } from './tokenProvider';
+import { MagicLinkTemplate as magicLinkTemplate } from '../templates';
+
+export class MagicLinkHandlers implements IAuthenticationStrategy {
+  private emailModule: Email;
+  private initialized: boolean = false;
+
+  constructor(private readonly grpcSdk: ConduitGrpcSdk) {}
+
+  async validate(): Promise<boolean> {
+    const config = ConfigController.getInstance().config;
+    if (config.magic_link.enabled && this.grpcSdk.isAvailable('email')) {
+      this.emailModule = this.grpcSdk.emailProvider!;
+      await this.registerTemplates();
+      return (this.initialized = true);
+    } else {
+      return (this.initialized = false);
+    }
+  }
+
+  async declareRoutes(routingManager: RoutingManager): Promise<void> {
+    const config = ConfigController.getInstance().config;
+    routingManager.route(
+      {
+        path: '/hook/magic-link',
+        action: ConduitRouteActions.POST,
+        description: `A webhook used to send a magic link to a user.`,
+        bodyParams: {
+          email: ConduitString.Required,
+        },
+      },
+      new ConduitRouteReturnDefinition('MagicLinkSendResponse', 'String'),
+      this.sendMagicLink.bind(this),
+    );
+
+    routingManager.route(
+      {
+        path: '/hook/magic-link/:verificationToken',
+        action: ConduitRouteActions.GET,
+        description: `A webhook used to verify a user who has received a magic link.`,
+        urlParams: {
+          verificationToken: ConduitString.Required,
+        },
+      },
+      new ConduitRouteReturnDefinition('VerifyMagicLinkLoginResponse', {
+        userId: ConduitString.Optional,
```

Comment:
align the returns with the rest of the authorize endpoints from other handlers.
i.e. userId is not returned, access tokens are both optional, message is not returned